### PR TITLE
Do not count multiple reviews from the same user

### PR DESCRIPTION
**Pull request by chrysn**
My two ACKs (long apart, second reaffirming) were counted as two in https://github.com/RIOT-OS/RIOT/pull/18942 for where two ACKs would have been needed.

I've added the mock by how I read https://pygithub.readthedocs.io/en/stable/github_objects/PullRequestReview.html and https://docs.github.com/en/rest/pulls/reviews?apiVersion=2022-11-28, is there any good way to confirm that .login is really a field access?

### CODE REVIEW

**Comment by chrysn** [comment]
For lack of rights to assign @miri64, pinging instead. Does this repo require extra privileges to kick off actions as well?

**Comment by miri64** [comment]
> is there any good way to confirm that .login is really a field access?

I just tested it in the console for https://github.com/RIOT-OS/RIOT/pull/20647 (which has three reviews):

```
In [1]: import github

In [2]: repo = github.Github("ghp_xxxxxxxxxxxxxxxxxxxxx").get_rep
   ...: o("RIOT-OS/RIOT")

In [3]: pull = repo.get_pull(20647)

In [4]: for review in pull.get_reviews():
   ...:     print(review.user.login)
   ...: 
mguetschow
dylad
Teufelchen1
```

**Comment by miri64** [comment]
> repo = github.Github("ghp_xxxxxxxxxxxxxxxxxxxxx").get_rep
>    ...: o("RIOT-OS/RIOT")

Wow Github is smart :D accidentally posted the token and immediately it was revoked.

**Review by miri64 — APPROVED** [review]
Code looks good and tests are passing. ACK!

**Comment by miri64** [comment]
> Does this repo require extra privileges to kick off actions as well?

Could you try to rebase? :-/

**Comment by chrysn** [comment]
Nothing to rebase; commit--amend'ed and force pushed for the desired effect, CI now running.

**Comment by miri64** [comment]
Just use `black` ;-)

**Comment by chrysn** [comment]
Force pushed a single lint fix ("expected 1 blank line before a nested definition, found 0"), the others might be still showing but be on some ignore list (I didn't touch the overly long lines).

If they show up again, I'll do the needful in a separate cleanup commit.

**Comment by miri64** [comment]
See https://github.com/RIOT-OS/check-labels-action/pull/3 for the `yield from` issue.

File: check_labels.py [hunk 1]
```diff
@@ -175,13 +175,13 @@ def parse_condition(condition):
...
     for review in pull.get_reviews():
         if review.state == "APPROVED":
-            approvals += 1
-            if approvals > condition[1]:
+            approving_reviewers.add(review.user.login)
+            if len(approving_reviewers) > condition[1]:
                 print(
-                    f"PR#{pull} has {approvals}/{condition[1] + 1} approvals,"
+                    f"PR#{pull} has {len(approving_reviewers)}/{condition[1] + 1} approving reviewers,"
                     f" removing label '{missing_approvals_label}'"
```
Comment (miri64) [inline check_labels.py]:
```suggestion
                    f"PR#{pull} has {len(approving_reviewers)}/{condition[1] + 1} "
                    f"approving reviewers, removing label '{missing_approvals_label}'"
```

File: check_labels.py [hunk 2]
```diff
@@ -197,7 +197,7 @@ def check_review_approvals(pull, condition, missing_approvals_label):
 
     if missing_approvals_label and condition[1] > 0:
         print(
-            f"PR#{pull} has only {approvals}/{condition[1] + 1} approvals,"
+            f"PR#{pull} has only {len(approving_reviewers)}/{condition[1] + 1} approving reviewers,"
             f" setting label '{missing_approvals_label}'"
```
Comment (miri64) [inline check_labels.py]:
```suggestion
            f"PR#{pull} has only {len(approving_reviewers)}/{condition[1] + 1} "
            f"approving reviewers, setting label '{missing_approvals_label}'"
```

**Comment by chrysn** [comment]
> Just use black ;-)

I did, and it was happy. Apparently, linters disagree with each other.

I'll fix this one manually, but urge you to stop making styling mandatory where the auto-indent tools don't fix things.

**Comment by miri64** [comment]
> I did, and it was happy. Apparently, linters disagree with each other.

Black does not honor line lengths for strings.



> I'll fix this one manually, but urge you to stop making styling mandatory where the auto-indent tools don't fix things.

I provided suggestions

File: check_labels.py (hunk 1 above)
Comment (chrysn) [reply check_labels.py]: Thanks, obsolete by last simultaneous commit.

**Comment by miri64** [comment]
Removed the requirement for the commits to be signed.

**Pull request merged**